packages/grafana-data/src/utils/datasource.ts

```diff
@@ -118,16 +118,13 @@ export const updateDatasourcePluginSecureJsonDataOption = <J, S extends {} = Key
   val: any
 ) => {
   const config = props.options;
-  if (!config.secureJsonData) {
-    return;
-  }
 
   props.onOptionsChange({
     ...config,
     secureJsonData: {
-      ...config.secureJsonData,
+      ...(config.secureJsonData ? config.secureJsonData : {}),
```

Comment:
in theory, using `as ...` was a bit of cheating but I cannot find a better a solution (I don't know if someone has a better suggestion). We can reduce the scope a bit just setting the type of the empty object:

```suggestion
      ...(config.secureJsonData ? config.secureJsonData : {} as S),
```